mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -242,29 +241,13 @@ pub async fn my_balance(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Stri
     Ok(try_s!(Response::builder().body(res)))
 }
 
-#[cfg(not(target_arch = "wasm32"))]
-async fn close_async_connection(ctx: &MmArc) {
-    if let Some(async_conn) = ctx.async_sqlite_connection.as_option() {
-        let mut conn = async_conn.lock().await;
-        if let Err(e) = conn.close().await {
-            error!("Error stopping AsyncConnection: {}", e);
-        }
-    }
-}
-
 pub async fn stop(ctx: MmArc) -> Result<Response<Vec<u8>>, String> {
     dispatch_lp_event(ctx.clone(), StopCtxEvent.into()).await;
     // Should delay the shutdown a bit in order not to trip the "stop" RPC call in unit tests.
     // Stopping immediately leads to the "stop" RPC call failing with the "errno 10054" sometimes.
     let fut = async move {
         Timer::sleep(0.05).await;
-
-        #[cfg(not(target_arch = "wasm32"))]
-        close_async_connection(&ctx).await;
-
-        if let Err(e) = ctx.stop() {
-            error!("Error stopping MmCtx: {}", e);
-        }
+        ctx.stop().await.expect("Couldn't stop the KDF runtime.");
```

Comment:
Related to the first comment, I see that you changed the error logging to `expect` here too. I guess you have a good reason to why expect is better, can you please elaborate on this?

P.S. Panicking will only terminate the thread we called stop from afaik, so if the reason for expect is that mm2 will not be in a right state if we don't panic, it will be the same with expect.

Reply:
> I guess you have a good reason to why expect is better, can you please elaborate on this?

By default, panicking triggers unwinding logic and cleans up the resources from the thread. But we have a custom panic strategy which I forgot about.. I will switch it back to the default so we get benefit from panicking.